git_machete/client.py

```diff
@@ -1283,46 +1284,47 @@ def rebase(self, onto: AnyRevision, from_exclusive: AnyRevision, branch: LocalBr
         print('Checking for unmanaged branches...')
-        branches_to_delete = excluding(self.__git.get_local_branches(), self.managed_branches)
-        self.__delete_branches(branches_to_delete=branches_to_delete, opt_yes=opt_yes)
+        branches_to_delete = sorted(excluding(self.__git.get_local_branches(), self.managed_branches))
+        self.__delete_branches(branches_to_delete=branches_to_delete,
+                               opt_squash_merge_detection=opt_squash_merge_detection, opt_yes=opt_yes)
 
-    def __delete_branches(self, branches_to_delete: List[LocalBranchShortName], opt_yes: bool) -> None:
+    def __delete_branches(self, branches_to_delete: List[LocalBranchShortName],
+                          opt_squash_merge_detection: SquashMergeDetection, opt_yes: bool) -> None:
         current_branch = self.__git.get_current_branch_or_none()
         if current_branch and current_branch in branches_to_delete:
             branches_to_delete = excluding(branches_to_delete, [current_branch])
             print(f"Skipping current branch {bold(current_branch)}")
-        if branches_to_delete:
-            branches_merged_to_head = self.__git.get_merged_local_branches()
-
-            branches_to_delete_merged_to_head = [branch for branch in branches_to_delete if branch in branches_merged_to_head]
-            for branch in branches_to_delete_merged_to_head:
-                remote_branch = self.__git.get_strict_counterpart_for_fetching_of_branch(branch)
-                is_merged_to_remote = self.__git.is_ancestor_or_equal(branch.full_name(),
-                                                                      remote_branch.full_name()) if remote_branch else True
-                msg_core_suffix = '' if is_merged_to_remote else f', but not merged to {bold(remote_branch)}'  # type: ignore[arg-type]
-                msg_core = f"{bold(branch)} (merged to HEAD{msg_core_suffix})"
-                msg = f"Delete branch {msg_core}?" + get_pretty_choices('y', 'N', 'q')
-                opt_yes_msg = f"Deleting branch {msg_core}..."
-                ans = self.ask_if(msg, opt_yes_msg, opt_yes=opt_yes)
-                if ans in ('y', 'yes'):
-                    self.__git.delete_branch(branch, force=is_merged_to_remote)
-                elif ans in ('q', 'quit'):
-                    return
+        if not branches_to_delete:
+            print("No branches to delete")
+            return
 
-            branches_to_delete_unmerged_to_head = [branch for branch in branches_to_delete if branch not in branches_merged_to_head]
-            for branch in branches_to_delete_unmerged_to_head:
-                msg_core = f"{bold(branch)} (unmerged to HEAD)"
+        if opt_yes:
+            for branch in branches_to_delete:
+                print(f"Deleting branch {bold(branch)}...")
+                self.__git.delete_branch(branch, force=True)
+        else:
```

Comment:
@cinnamennen work in progress... note that this will change the message displayed next to the squash-merged branch (from `unmerged` to `merged`), but the confirmation will still be needed (i.e. `--yes` is needed to turn off the need  for confirmation completely)